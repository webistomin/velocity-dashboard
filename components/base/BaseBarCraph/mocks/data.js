export const barGraphData = {
  labels: ['April', 'May', 'June'],
  datasets: [
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Comfort',
      lineTension: 0.4,
      backgroundColor: '#2E5BFF',
      data: [60, 65, 85],
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Premium',
      backgroundColor: '#8C54FF',
      data: [70, 80, 90],
    },
  ],
};
