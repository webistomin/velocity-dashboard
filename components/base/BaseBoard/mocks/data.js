export const boardData = [
  {
    title: 'Backlog',
    color: 'blue',
    tasks: [
      {
        title: 'Spire1',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire2',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire3',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'In progress',
    color: 'purple',
    tasks: [
      {
        title: 'Spire5',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire6',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire7',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire8',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Validate',
    color: 'darkturquoise',
    tasks: [
      {
        title: 'Spire9',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire10',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire11',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire12',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Complete',
    color: 'yellow',
    tasks: [
      {
        title: 'Spire13',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire14',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire15',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire16',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
];
