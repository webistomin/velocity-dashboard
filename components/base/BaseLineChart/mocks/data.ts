import { ChartData } from 'chart.js';

export const lineGraphData: ChartData = {
  labels: [
    '',
    'Mar 1',
    'Mar 8',
    'Mar 15',
    'Mar 22',
    'Mar 29',
    'Apr 5',
    'Apr 12',
    'Apr 19',
    'Apr 26',
    'May 2',
    'May 9',
    'May 16',
    'May 23',
    'May 30',
    'Jun 5',
    'Jun 10',
    'Jun 19',
  ],
  datasets: [
    {
      label: 'Revenue',
      borderColor: '#2E5BFF',
      pointBackgroundColor: '#FFF',
      fill: true,
      lineTension: 0,
      data: [
        {
          x: 0,
          y: 8000,
        },
        {
          x: 1,
          y: 11500,
        },
        {
          x: 2,
          y: 9800,
        },
        {
          x: 3,
          y: 15100,
        },
        {
          x: 4,
          y: 23000,
        },
        {
          x: 5,
          y: 22000,
        },
        {
          x: 6,
          y: 16000,
        },
        {
          x: 7,
          y: 13800,
        },
        {
          x: 8,
          y: 16300,
        },
        {
          x: 9,
          y: 17000,
        },
        {
          x: 10,
          y: 24500,
        },
        {
          x: 11,
          y: 19000,
        },
        {
          x: 12,
          y: 12000,
        },
        {
          x: 13,
          y: 8900,
        },
        {
          x: 14,
          y: 16000,
        },
        {
          x: 15,
          y: 27000,
        },
        {
          x: 16,
          y: 22000,
        },
        {
          x: 17,
          y: 21000,
        },
      ],
    },
    {
      label: 'Trips',
      borderColor: '#8C54FF',
      pointBackgroundColor: '#FFF',
      fill: true,
      lineTension: 0,
      data: [
        {
          x: 0,
          y: 4800,
        },
        {
          x: 1,
          y: 5000,
        },
        {
          x: 2,
          y: 7500,
        },
        {
          x: 3,
          y: 7000,
        },
        {
          x: 4,
          y: 11500,
        },
        {
          x: 5,
          y: 13000,
        },
        {
          x: 6,
          y: 7500,
        },
        {
          x: 7,
          y: 8500,
        },
        {
          x: 8,
          y: 6000,
        },
        {
          y: 13700,
        },
        {
          x: 10,
          y: 18000,
        },
        {
          x: 11,
          y: 17600,
        },
        {
          x: 12,
          y: 2500,
        },
        {
          x: 13,
          y: 6000,
        },
        {
          x: 14,
          y: 9000,
        },
        {
          x: 15,
          y: 22500,
        },
        {
          x: 16,
          y: 14000,
        },
        {
          x: 17,
          y: 19000,
        },
      ],
    },
  ],
};
