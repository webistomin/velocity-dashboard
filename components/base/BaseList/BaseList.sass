+b('base-list')
  counter-reset: base-list

  +e('item')
    width: 100%
    display: flex
    counter-increment: base-list

    &:not(:last-of-type)
      margin-bottom: rem(14px)

  +e('col')
    +m('right')
      width: 100%

    +m('left')
      position: relative
      width: rem(48px)
      margin-right: rem(16px)

      &::before
        content: counters(base-list, ".") " "
        width: rem(16px)
        height: rem(16px)
        display: flex
        align-items: center
        justify-content: center
        position: absolute
        top: 0
        right: 0
        background-color: var(--color-white)
        font: var(--text-counter)
        color: var(--color-gray-3)
        border-radius: 50%
        z-index: 1
        box-shadow: 0 rem(3px) rem(10px) var(--color-lavender)

  +e('row')
    display: flex
    justify-content: space-between
    align-items: baseline

  +e('name')
    font-weight: 400

  +e('text')
    font-weight: 400
    color: var(--color-gray-3)

  +e('stat')
    color: var(--color-gray-3)
