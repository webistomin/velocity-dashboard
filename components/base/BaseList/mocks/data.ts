export const listData = [
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
];
