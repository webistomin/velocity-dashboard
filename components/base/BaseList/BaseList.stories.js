import { object, number } from '@storybook/addon-knobs';
import BaseList from './BaseList';
import { listData } from './mocks/data';

export default {
  title: 'BaseList',
  parameters: {
    options: {
      showPanel: true,
      selectedPanel: 'storybook/knobs/panel',
    },
  },
};

export const Default = () => ({
  components: { BaseList },
  props: {
    list: {
      type: Array,
      default: object('list', listData),
      required: true,
    },
    limit: {
      type: Number,
      default: number('limit', 6),
    },
  },
  template: `<base-list
    :list='this.list'
    :limit='this.limit'
  />`,
});
