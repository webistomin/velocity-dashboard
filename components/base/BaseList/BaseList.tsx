import { VueComponent } from 'types/vue-components';
import { Component, Prop } from 'nuxt-property-decorator';

import BaseThumbnail from 'components/base/BaseThumbnail';

import './BaseList.sass';

export interface IBaseList {
  image?: string;
  name: string;
  text: string;
  statistic: string;
  price: string;
}

export interface IBaseListProps {
  list: IBaseList[];
  limit?: number;
}

@Component({
  name: 'BaseList',
})
export default class BaseList extends VueComponent<IBaseListProps> {
  @Prop()
  private readonly list!: IBaseListProps['list'];

  @Prop({ default: 6 })
  private readonly limit!: IBaseListProps['limit'];

  render() {
    return (
      <div class='base-list'>
        <ul class='base-list__list list'>
          {this.list.slice(0, this.limit).map((item, index) => {
            return (
              <li class='base-list__item list-item' key={`item-${index}`}>
                <div class='base-list__col base-list__col_left'>
                  <BaseThumbnail size='m' image={item.image} alt={item.name} />
                </div>
                <div class='base-list__col base-list__col_right'>
                  <div class='base-list__row'>
                    <strong class='base-list__name'>{item.name}</strong>
                    <span class='base-list__price'>{item.price}</span>
                  </div>
                  <div class='base-list__row'>
                    <strong class='base-list__text'>{item.text}</strong>
                    <span class='base-list__stat'>{item.statistic}</span>
                  </div>
                </div>
              </li>
            );
          })}
        </ul>
      </div>
    );
  }
}
