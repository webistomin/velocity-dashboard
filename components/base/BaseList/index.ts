import BaseList from './BaseList';

export default BaseList;
