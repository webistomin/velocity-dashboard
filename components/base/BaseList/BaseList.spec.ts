import { mount } from '@vue/test-utils';
import BaseList from './BaseList';
import { listData } from './mocks/data';

describe('BaseList', () => {
  const factory = (propsData = {}, options = {}) => {
    return mount(BaseList, {
      propsData: {
        ...propsData,
        list: listData,
      },
      stubs: ['v-lazy-image'],
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });
});
