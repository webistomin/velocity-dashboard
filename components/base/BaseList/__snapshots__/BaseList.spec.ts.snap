// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseList Matches snapshot 1`] = `
"<div class=\\"base-list\\">
  <ul class=\\"base-list__list list\\">
    <li class=\\"base-list__item list-item\\">
      <div class=\\"base-list__col base-list__col_left\\"><span class=\\"base-thumbnail base-thumbnail_size_m   \\"><picture class=\\"base-thumbnail__picture picture\\"><v-lazy-image-stub src=\\"/img/avatar.png\\" alt=\\"Bebop\\" class=\\"base-thumbnail__img image\\"></v-lazy-image-stub></picture></span></div>
      <div class=\\"base-list__col base-list__col_right\\">
        <div class=\\"base-list__row\\"><strong class=\\"base-list__name\\">Bebop</strong><span class=\\"base-list__price\\">$6,432</span></div>
        <div class=\\"base-list__row\\"><strong class=\\"base-list__text\\">Volvo Intellisafe</strong><span class=\\"base-list__stat\\">1,232 miles</span></div>
      </div>
    </li>
    <li class=\\"base-list__item list-item\\">
      <div class=\\"base-list__col base-list__col_left\\"><span class=\\"base-thumbnail base-thumbnail_size_m   \\"><picture class=\\"base-thumbnail__picture picture\\"><v-lazy-image-stub src=\\"/img/avatar.png\\" alt=\\"Bebop\\" class=\\"base-thumbnail__img image\\"></v-lazy-image-stub></picture></span></div>
      <div class=\\"base-list__col base-list__col_right\\">
        <div class=\\"base-list__row\\"><strong class=\\"base-list__name\\">Bebop</strong><span class=\\"base-list__price\\">$6,432</span></div>
        <div class=\\"base-list__row\\"><strong class=\\"base-list__text\\">Volvo Intellisafe</strong><span class=\\"base-list__stat\\">1,232 miles</span></div>
      </div>
    </li>
    <li class=\\"base-list__item list-item\\">
      <div class=\\"base-list__col base-list__col_left\\"><span class=\\"base-thumbnail base-thumbnail_size_m   \\"><picture class=\\"base-thumbnail__picture picture\\"><v-lazy-image-stub src=\\"/img/avatar.png\\" alt=\\"Bebop\\" class=\\"base-thumbnail__img image\\"></v-lazy-image-stub></picture></span></div>
      <div class=\\"base-list__col base-list__col_right\\">
        <div class=\\"base-list__row\\"><strong class=\\"base-list__name\\">Bebop</strong><span class=\\"base-list__price\\">$6,432</span></div>
        <div class=\\"base-list__row\\"><strong class=\\"base-list__text\\">Volvo Intellisafe</strong><span class=\\"base-list__stat\\">1,232 miles</span></div>
      </div>
    </li>
    <li class=\\"base-list__item list-item\\">
      <div class=\\"base-list__col base-list__col_left\\"><span class=\\"base-thumbnail base-thumbnail_size_m   \\"><picture class=\\"base-thumbnail__picture picture\\"><v-lazy-image-stub src=\\"/img/avatar.png\\" alt=\\"Bebop\\" class=\\"base-thumbnail__img image\\"></v-lazy-image-stub></picture></span></div>
      <div class=\\"base-list__col base-list__col_right\\">
        <div class=\\"base-list__row\\"><strong class=\\"base-list__name\\">Bebop</strong><span class=\\"base-list__price\\">$6,432</span></div>
        <div class=\\"base-list__row\\"><strong class=\\"base-list__text\\">Volvo Intellisafe</strong><span class=\\"base-list__stat\\">1,232 miles</span></div>
      </div>
    </li>
    <li class=\\"base-list__item list-item\\">
      <div class=\\"base-list__col base-list__col_left\\"><span class=\\"base-thumbnail base-thumbnail_size_m   \\"><picture class=\\"base-thumbnail__picture picture\\"><v-lazy-image-stub src=\\"/img/avatar.png\\" alt=\\"Bebop\\" class=\\"base-thumbnail__img image\\"></v-lazy-image-stub></picture></span></div>
      <div class=\\"base-list__col base-list__col_right\\">
        <div class=\\"base-list__row\\"><strong class=\\"base-list__name\\">Bebop</strong><span class=\\"base-list__price\\">$6,432</span></div>
        <div class=\\"base-list__row\\"><strong class=\\"base-list__text\\">Volvo Intellisafe</strong><span class=\\"base-list__stat\\">1,232 miles</span></div>
      </div>
    </li>
    <li class=\\"base-list__item list-item\\">
      <div class=\\"base-list__col base-list__col_left\\"><span class=\\"base-thumbnail base-thumbnail_size_m   \\"><picture class=\\"base-thumbnail__picture picture\\"><v-lazy-image-stub src=\\"/img/avatar.png\\" alt=\\"Bebop\\" class=\\"base-thumbnail__img image\\"></v-lazy-image-stub></picture></span></div>
      <div class=\\"base-list__col base-list__col_right\\">
        <div class=\\"base-list__row\\"><strong class=\\"base-list__name\\">Bebop</strong><span class=\\"base-list__price\\">$6,432</span></div>
        <div class=\\"base-list__row\\"><strong class=\\"base-list__text\\">Volvo Intellisafe</strong><span class=\\"base-list__stat\\">1,232 miles</span></div>
      </div>
    </li>
  </ul>
</div>"
`;
